sound/soc/sof/ipc4.c

```diff
@@ -601,6 +601,31 @@ static int ipc4_fw_ready(struct snd_sof_dev *sdev, struct sof_ipc4_msg *ipc4_msg
 	return sof_ipc4_init_msg_memory(sdev);
 }
 
+static void sof_ipc4_setup_exception_debugfs_node(struct snd_sof_dev *sdev)
+{
+	struct snd_sof_dfsentry *dfse;
+	u32 telemetry_data_offset;
+	void __iomem *base_addr;
+
+	telemetry_data_offset =
+		sof_ipc4_find_debug_slot_offset_by_type(sdev,
+							SOF_IPC4_DEBUG_SLOT_TELEMETRY);
+	if (!telemetry_data_offset)
+		return;
+
+	base_addr = (void __iomem *)(telemetry_data_offset + sizeof(u32));
+	list_for_each_entry(dfse, &sdev->dfsentry_list, list) {
+		/* nothing to do if exception debugfs node exists */
+		if (dfse->io_mem == base_addr)
```

Comment:
By definition (and not by SOF implementation) the slot allocation is dynamic. In theory it is a possibility that the TELEMETRY slot is moving around, the base_addr will be different but the `exception` file is already there.
Should we look for the file also, just to make sure that it is not going to be the case and remove/recreate it (or just update the io_mem of the existing one?